packages/web3-eth-ens/src/registry.ts

```diff
@@ -0,0 +1,211 @@
+import { inputAddressFormatter, ReceiptInfo, RevertInstructionError } from 'web3-common';
+import { Contract, NonPayableCallOptions, TransactionReceipt } from 'web3-eth-contract';
+import { Address, isHexStrict, sha3Raw } from 'web3-utils';
+import { REGISTRY as registryABI } from './ABI/Registry';
+import { RESOLVER as resolverABI } from './ABI/Resolver';
+import { namehash } from './utils';
+
+export class Registry {
+	private readonly contract: Contract<typeof registryABI>;
+
+	public constructor() {
+		// TODO for contract, when eth.net is finished we can check network
+		this.contract = new Contract(registryABI);
+	}
+	public async getOwner(name: string) {
+		const promise = new Promise<[string]>((resolve, reject) => {
+			this.contract.methods
+				.owner(namehash(name))
+				.call()
+				.then(owner => {
+					resolve(owner);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async setOwner(
+		name: string,
+		address: Address,
+		txConfig: NonPayableCallOptions,
+	): Promise<ReceiptInfo | RevertInstructionError> {
+		const promise = new Promise<ReceiptInfo>((resolve, reject) => {
+			this.contract.methods
+				.setOwner(namehash(name), inputAddressFormatter(address))
+				.send(txConfig)
+				.then(receipt => {
+					resolve(receipt);
+				})
+				.catch(() => {
+					reject(RevertInstructionError);
+				});
+		});
+		return promise;
+	}
+
+	public async getTTL(name: string) {
+		const promise = new Promise((resolve, reject) => {
+			this.contract.methods
+				.getTTL(namehash(name))
```

Comment:
I think you should directly call ttl as: https://github.com/ensdomains/ens/blob/ff0f41747c05f1598973b0fe7ad0d9e09565dfcd/contracts/ENS.sol#L29

```suggestion
				.ttl(namehash(name))
```